sympy/functions/combinatorial/numbers.py

```diff
@@ -424,6 +424,9 @@ def _bell_incomplete_poly(n, k, symbols):
 
     @classmethod
     def eval(cls, n, k_sym=None, symbols=None):
+        if n is S.Infinity:
```

Comment:
`if n is S.Infinity and k_sym is None:`  will better handle the case of `bell(oo,x)`